components/tikv_util/src/metrics/process_linux.rs

```diff
@@ -21,45 +21,51 @@ pub fn monitor_process() -> Result<()> {
 pub struct ProcessCollector {
     pid: libc::pid_t,
     descs: Vec<Desc>,
-    cpu_total: Mutex<Counter>,
-    vsize: Gauge,
-    rss: Gauge,
-    start_time: Gauge,
+    cpu_total: Mutex<IntCounter>,
+    vsize: IntGauge,
+    rss: IntGauge,
+    start_time: IntGauge,
 }
 
 impl ProcessCollector {
     pub fn new(pid: libc::pid_t) -> Self {
         let mut descs = Vec::new();
 
-        let cpu_total = Counter::with_opts(Opts::new(
+        let cpu_total = IntCounter::with_opts(Opts::new(
             "process_cpu_seconds_total",
             "Total user and system CPU time spent in \
                  seconds.",
         ))
         .unwrap();
         descs.extend(cpu_total.desc().into_iter().cloned());
 
-        let vsize = Gauge::with_opts(Opts::new(
+        let vsize = IntGauge::with_opts(Opts::new(
             "process_virtual_memory_bytes",
             "Virtual memory size in bytes.",
         ))
         .unwrap();
         descs.extend(vsize.desc().into_iter().cloned());
 
-        let rss = Gauge::with_opts(Opts::new(
+        let rss = IntGauge::with_opts(Opts::new(
             "process_resident_memory_bytes",
             "Resident memory size in bytes.",
         ))
         .unwrap();
         descs.extend(rss.desc().into_iter().cloned());
 
-        let start_time = Gauge::with_opts(Opts::new(
+        let start_time = IntGauge::with_opts(Opts::new(
             "process_start_time_seconds",
             "Start time of the process since unix epoch \
                  in seconds.",
         ))
         .unwrap();
         descs.extend(start_time.desc().into_iter().cloned());
+        // proc_start_time
+        if let Some(boot_time) = *BOOT_TIME {
+            if let Ok(p) = procfs::process::Process::new(pid) {
+                start_time.set(p.stat.starttime as i64 / *CLK_TCK + boot_time as i64);
```

Comment:
Why not simply call `procfs::boot_time_secs()` here?

Reply:
Thanks, the `static BOOT_TIME` only call once, doesn't need to be static